### DISCUSSION TITLE
The status of RTSP stream flapping between "Preparing" and "Offline" even that RTSP stream available normally

### DISCUSSION THREAD

**Discussion by witchaphongli** [ant-media/Ant-Media-Server#6885] [Q&A] [answered]
### Short description
I just updated version to 2.11.3 20240906_0602 but we face an issue with RTSP stream can not start. It keeps flapping between "preparing" and "offline" (as following attached video) while RTMP streams are work as usual. 

https://github.com/user-attachments/assets/0c351092-785c-4858-ae04-a936be2b3d84

However I confirm that the target RTSP stream work properly by using VLC, the stream is display as normally.

### Environment

* Operating system and version: 
> * Ubuntu 20.04
* Java version: 
> - openjdk version "17.0.13" 2024-10-15
> - OpenJDK Runtime Environment (build 17.0.13+11-Ubuntu-2ubuntu120.04)
> - OpenJDK 64-Bit Server VM (build 17.0.13+11-Ubuntu-2ubuntu120.04, mixed mode, sharing)
* Ant Media Server version: 
> * Enterprise Edition 2.11.3 20240906_0602
* Browser name and version: 
> * Brave browser
* Server resource: 
<img width="1440" alt="Screenshot 2567-12-16 at 14 22 51" src="https://github.com/user-attachments/assets/9cb95088-9e4e-405b-8b93-eb7a30a48aa3" />
> * Topology
> * AntMedia server is behind firewall with NAT with 1 public IP address where **I set allow ALL for inbound/outbound on firewall** already to avoid any network relate issue.

### Steps to reproduce
1. Crate "New Stream Source" in particular "AppName" with Name/Stream URL/Stream Id
2. Click "Start broadcast"
3. Then the stream flapping between "preparing" and "offline"

### Expected behavior

That RTSP should become broadcast status on AntMedia UI as I try to Transcode Stream with `ffmpeg` from RTSP to RTMP that stream become broadcasting as normally

`# ffmpeg -rtsp_transport tcp -i 'rtsp://username:password@remote.rpro.digitalsmart.city:20049/Streaming/channels/102' -c:v copy -f flv rtmp://streaming.url.com/APPNAME/ffc7e0bd-c094-493f-b7be-xxxxxxxxxxxx`

the stream become broadcasting as normally (through RTMP)

https://github.com/user-attachments/assets/ec071f20-0549-47ac-90ca-5f483c51572c

### Actual behavior
I have try with `ffmpeg` and `ffprobe`

- **without** `-rtsp_transport tcp` option
`# ffmpeg -i 'rtsp://username:password@remote.rpro.digitalsmart.city:20049/Streaming/channels/102'`

![Screenshot 2567-12-14 at 16 15 14](https://github.com/user-attachments/assets/46b5b437-99a7-426e-b735-94d0d43ad170)

- **with** `-rtsp_transport tcp` option
`# ffmpeg -rtsp_transport tcp -i 'rtsp://admin:fb0b9v]F:]^=yjo@remote.rpro.digitalsmart.city:20049/Streaming/channels/102'`

![Screenshot 2567-12-14 at 16 36 48](https://github.com/user-attachments/assets/10bf26c4-f83e-4ef0-8fa3-bc13b78c608a)

- **ffprobe**
`# ffprobe -rtsp_transport tcp -i 'rtsp://username:password@remote.rpro.digitalsmart.city:20049/Streaming/channels/102'`

![Screenshot 2567-12-14 at 16 32 45](https://github.com/user-attachments/assets/26ef8a40-ed76-4955-888b-7cbe15f555eb)

- **convert to .mp4**
`# ffmpeg -rtsp_transport tcp -i 'rtsp://username:password@remote.rpro.digitalsmart.city:20049/Streaming/channels/102' -c:v copy output.mp4`

![Screenshot 2567-12-14 at 17 07 13](https://github.com/user-attachments/assets/d2014c42-7694-4cf2-93e0-b8bff8fe97ff)

- **convert to .m3u8**
`# ffmpeg -rtsp_transport tcp -i 'rtsp://username:password@remote.rpro.digitalsmart.city:20049/Streaming/channels/102' -f hls output.m3u8`

![Screenshot 2567-12-14 at 17 11 09](https://github.com/user-attachments/assets/bc8d8aa5-cffb-4798-a52f-f09629f3a56e)

### Logs
- ant-media-server.log

![Screenshot 2567-12-14 at 15 52 08](https://github.com/user-attachments/assets/7e57c298-b149-4ecf-9d85-a0fecae005ef)

- antmedia-error.log

![Screenshot 2567-12-14 at 15 58 41](https://github.com/user-attachments/assets/3fc5298c-7809-4e44-ac44-44ab02e4c70c)

Here is my `red5.properties` file (which is the original from installation)
[red5.properties.txt](https://github.com/user-attachments/files/18147993/red5.properties.txt)

Any suggestion on this ? /\

**Comment by Mohit-3196** [comment, marked as answer]
Hi @witchaphongli 
How are you?

Can you please try the below settings under Advanced settings & recreate the affected RTSP stream?
`"rtspPullTransportType": "2",`

Let me know if that resolves the issue for you.

--
Cheers,
Mohit

**Reply by witchaphongli** [reply]
Hi @Mohit-3196 

Thank you for your suggestion!  May I confirm where to add that setting between
  `/usr/local/antmedia/webapps/APPNAME/WEB-INF/red5-web.properties ` or  `/usr/local/antmedia/conf/red5.properties `
  
  SInce I can not found where to add that setting in "Advance setting"
  
<img width="1163" alt="Screenshot 2567-12-18 at 21 44 37" src="https://github.com/user-attachments/assets/eb91cf32-57a2-45aa-9f6b-a7ae4e348871" />

**Reply by Mohit-3196** [reply]
Hi @witchaphongli,
If you are running v2.11.3, the advanced settings can be found in the application settings.
Please check [this](https://antmedia.io/docs/version-2.11.3/guides/configuration-and-testing/ams-application-configuration/#management-panel-settings). 

Cheers

**Reply by witchaphongli** [reply]
Dear @Mohit-3196 

Thank you for your suggestion! I surprised that, I never noticed that basic/advance setting toggle button on the top-right conner.
Luckily! I seems solve my issue,I have changed:

1. "rtspPullTransportType": "3" to "rtspPullTransportType": "2" and save
2. restart AntMedia service
3. Delete and recreate that issued RTSP stream.

<img width="1414" alt="Screenshot 2567-12-19 at 14 42 55" src="https://github.com/user-attachments/assets/ce5ec7ea-91ea-4b90-9ae5-321003d02bb8" />

All of them are RTSP stream. They are broadcasting now!
<img width="985" alt="Screenshot 2567-12-19 at 16 25 00" src="https://github.com/user-attachments/assets/74c6552c-4caf-49b0-a26e-ae75bdb5b0e7" />

I also revert my firewall policy back to UDP for RTSP. Also the RTMP stream still work as usual. 

I was struggling  this point for vary long time until your answer! Thank you for your help 

PS. Could you please explain why "rtspPullTransportType": "3" is not work. I understand that 3 is support both TCP and UDP

**Reply by Mohit-3196** [reply]
Thank you for the confirmation @witchaphongli,
Great to know that it's working for you now :)

> restart AntMedia service
You do not have to restart the Ant Media Server when changing the settings from the Web Panel. Just restart the stream & the settings will be applied.

Also, you can check the property & the values it can take from [here ](https://antmedia.io/javadoc/io/antmedia/AppSettings.html#rtspPullTransportType)

--
Cheers

**Comment by witchaphongli** [comment]
The issue is solved as comment then this good to close. Thank community!

**Reply by Mohit-3196** [reply]
Thank you @witchaphongli,
cheers.

**Comment by burak-58** [comment]
Great job @witchaphongli and @Mohit-3196. Great reporting and great support.
